service/src/main/java/tennisclub/service/RankingServiceImpl.java

```diff
@@ -0,0 +1,49 @@
+package tennisclub.service;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import tennisclub.dao.RankingDao;
+import tennisclub.entity.Tournament;
+import tennisclub.entity.User;
+import tennisclub.entity.ranking.Ranking;
+
+import java.util.List;
+
+@Service
+public class RankingServiceImpl implements RankingService {
+
+    private final RankingDao rankingDao;
+
+    @Autowired
+    RankingServiceImpl(RankingDao rankingDao) { this.rankingDao = rankingDao; }
+
+    @Override
+    public void create(Ranking ranking) {
+        rankingDao.create(ranking);
+    }
+
+    @Override
+    public Ranking find(Tournament tournament, User player) {
+        return rankingDao.find(tournament, player);
+    }
+
+    @Override
+    public List<Ranking> findByPlayer(User player) {
+        return rankingDao.findByUser(player);
+    }
+
+    @Override
+    public List<Ranking> findByTournament(Tournament tournament) {
+        return rankingDao.findByTournament(tournament);
+    }
+
+    @Override
+    public void update(Ranking ranking) {
```

Comment:
I think the update method should return the updated object, however I am not sure about this myself 😄

Reply:
Makes sense as the DAO update returns a new entity, thanks! I'll have to fix this for the user as well 😄